Civil-Service-Human-Resources/rpg-candidate-interface-ui
# [WIP] Improve the security of the application

**DO NOT MERGE YET -WIP**

__What?__

The context of this pull request is to focus on updating the security of the express/node.js application. We want to ensure that the web application is following the latest security practice in setting the correct HTTP headers, restricting the amount of information divulged and that our dependencies are up to date.


__How to test__

* Checkout the `develop` branch
* Run `npm test` and check that the tests fail 
* Check out this branch
* Run `npm test` to check that the tests pass 
* Break one of the tests
* Try to commit said test file 
* Ensure that you are stopped from committing due to pre-commit hook.
* Run round and give @dhilton a high five to ensure you have read these test steps.

__Who can test__

Anyone please, anyone who realises how bad it is the tests have been broken for so long and that I gave up my Sunday evening to fix them.
